Improve MTP coverage doc (#44522)
dotnet/docs: docs/core/testing/unit-testing-platform-extensions-code-coverage.md

| Option                     | Description                                                                   |
|----------------------------|-------------------------------------------------------------------------------|
| `--coverage`               | Collect the code coverage using dotnet-coverage tool.                         |
| `--coverage-output`        | The name or path of the produced coverage file. By default, the file is `TestResults/<guid>.coverage`. |
| `--coverage-output-format` | Output file format. Supported values are: `coverage`, `xml`, and `cobertura`. Default is `coverage`. |

> | `--coverage-settings`      | [XML code coverage settings](https://learn.microsoft.com/dotnet/core/additional-tools/dotnet-coverage#settings). |

Reviewer: ```suggestion
| `--coverage-settings`      | [XML code coverage settings](../additional-tools/dotnet-coverage.md#settings). |
```

For more information about the available options, see [settings](../additional-tools/dotnet-coverage.md#settings) and [samples](https://github.com/microsoft/codecoverage/tree/main/samples/Algorithms).

## Coverlet

> [!IMPORTANT]

> > The `coverlet.collector` NuGet package is designed specifically for VSTest and cannot be used with `Microsoft.Testing.Platform`

Reviewer: ```suggestion
> The `coverlet.collector` NuGet package is designed specifically for VSTest and cannot be used with `Microsoft.Testing.Platform`.
```

There's currently no Coverlet extension, but you can use [Coverlet .NET global tool](https://github.com/coverlet-coverage/coverlet#net-global-tool-guide-suffers-from-possible-known-issue).

> Assuming you have already installed coverlet global tool, you can now run:

Reviewer: ```suggestion
Assuming you've already installed the Coverlet global tool, you can now run:
```

```bash

> coverlet .\bin\Debug\net8.0\TestProject2.dll --target "dotnet" --targetargs "test .\bin\Debug\net8.0\TestProject2.dll --no-build"

Reviewer 2: Is it expected to pass the dll instead of exe to `dotnet test`?

Author: Both should work
